src/backend/distributed/executor/multi_utility.c

```diff
@@ -1255,7 +1255,13 @@ ExecuteDistributedDDLCommand(Oid relationId, const char *ddlCommandString,
 {
 	bool executionOK = false;
 
-	PreventTransactionChain(isTopLevel, "distributed DDL commands");
+	if (IsInTransactionChain(isTopLevel) && (XactModification > XACT_MODIFICATION_NONE))
```

Comment:
That's a good point. I'm not sure if this condition was entirely true before, though now the `IsInTransactionChain` is now superfluous. Taking your `Assert` suggestion.